### PR TITLE
feat: New Component Avatar Group [HOMER-2118]

### PR DESCRIPTION
**Pull request by Lelith**
# Purpose of PR

Introduces a new component for grouping Avatars. It comes in two sizes and two different designs (stacked and spaced). It will show a maximum of 3 Avatar children, the rest of the children get rendered as a Menu. 
The Avatar components require an alt-text which will be displayed as the user name.

image

image

### CODE REVIEW

**Review by mdmjg — APPROVED** [review]

File: packages/components/avatar/README.mdx [hunk 1]
```diff
@@ -97,3 +97,39 @@ When the avatar image is loading, a loading skeleton will be shown automatically
...
+### Sizes
+
+The group can be rendered in two different sizes
+
+- **Small** - 24px
+- **Medium** - 32px, default size
+
+```jsx file=./examples/AvatarGroupSizeExample.tsx
+
+```
```
Comment (miguelcrespo) [inline packages/components/avatar/README.mdx:135]: question: based on the example of the `Avatar` shouldn't this headings be under `## Examples` ?

It also doesn't have the `import` section

**Review by miguelcrespo — APPROVED** [review]
I also don't see the new component in the preview: https://forma-36-git-feat-avatargrouphomer-2118.colorfuldemo.com/components/avatar

Is this expected?

**Review by denkristoffer — CHANGES_REQUESTED** [review]
The docs are broken: https://forma-36-git-feat-avatargrouphomer-2118.colorfuldemo.com/components/avatar

At least one of the errors is because the new component has not been added to the SandpackRenderer and LiveEditor.

File: packages/components/avatar/src/AvatarGroup/AvatarGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import { Flex, type CommonProps } from '@contentful/f36-core';
+import { Menu } from '@contentful/f36-menu';
+import type { SpacingTokens } from '@contentful/f36-tokens';
+import { type AvatarProps } from '../Avatar';
+import { getAvatarGroupStyles } from './AvatarGroup.styles';
+
+import { cx } from 'emotion';
+
+export interface AvatarGroupProps extends CommonProps {
+  spacing?: SpacingTokens;
```
Comment (denkristoffer) [inline packages/components/avatar/src/AvatarGroup/AvatarGroup.tsx:11]: I don't see the `spacing` prop being used anywhere, should it be?

File: packages/components/avatar/src/AvatarGroup/AvatarGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+  const childrenArray = React.Children.toArray(children);
+  const childrenToRenderCount = childrenArray.length > 3 ? 2 : 3;
+  const childrenToRender = childrenArray.slice(0, childrenToRenderCount);
+  const childrenInMenu = childrenArray.slice(childrenToRenderCount);
+
+  return (
+    <Flex
+      flexDirection="row"
+      data-test-id={testId}
```
Comment (denkristoffer) [inline packages/components/avatar/src/AvatarGroup/AvatarGroup.tsx:39]:
```suggestion
      testId={testId}
```

**Comment by denkristoffer** [comment]
Can you please bump the package.json version? You can do it with `npm run-script prerelease` or manually.

**Comment by miguelcrespo** [comment]
> Can you please bump the package.json version? You can do it with `npm run-script prerelease` or manually.

@denkristoffer Done

**Review by denkristoffer — APPROVED** [review]

**Pull request merged**